event_detector_scripts/delete-event-detectors.js

```diff
@@ -74,34 +75,38 @@ let count = 0;
 let failed = 0;
 for (const eventDetectorCsv of eventDetectorsArray) {
 
-    //validations
-    try {
-        for (const [key, value] of Object.entries(eventDetectorCsv)) {
-            if (mainHeaders.includes(key)) {
-                assureNotEmpty(key, value);
-            }
-        }
+    let detector;
+
+    //VALIDATIONS
 
-    } catch (error) {
-        log.error('Will not create new Event Detector. {} ', error.message);
-        verbose(error.message);
+    //Confirm the ID and XID values in the CSV are set
+    if (!changeProperty(eventDetectorCsv.eventDetectorId)) {
+        log.error('Event detector ID not provided. CSV contains: {}', eventDetectorCsv.eventDetectorId);
+        verbose(`Event detector ID not provided. CSV contains: ${eventDetectorCsv.eventDetectorId}`);
         failed++;
         continue;
     }
-
-    let detector;
+    if (!changeProperty(eventDetectorCsv.eventDetectorXid)) {
+        log.error('Event detector XID not provided. CSV contains: {}', eventDetectorCsv.eventDetectorXid);
+        verbose(`Event detector XID not provided. CSV contains: ${eventDetectorCsv.eventDetectorXid}`);
+        failed++;
+        continue;
+    }
+    
+    //Confirm the event detector XID exists
+    verbose(eventDetectorCsv.eventDetectorXid);
     try {
         detector = eventDetectorsService.get(eventDetectorCsv.eventDetectorXid);
     } catch (e) {
-        log.error('Failed deleting the event detector with XID {} NOT FOUND!', eventDetectorCsv.eventDetectorXid);
-        verbose(`'Failed deleting the event detector with XID ${eventDetectorCsv.eventDetectorXid} NOT FOUND!'`);
+        log.error('Failed finding the event detector with XID {}: {}', eventDetectorCsv.eventDetectorXid, e);
+        verbose(`Failed finding the event detector with XID ${eventDetectorCsv.eventDetectorXid}: ${e}`);
         failed++;
         continue;
     }
 
     try {
         compare(detector.getId(), eventDetectorCsv.eventDetectorId, "eventDetectorId MISMATCH");
```

Comment:
```suggestion
        compare(detector.getId(), Number(eventDetectorCsv.eventDetectorId), "eventDetectorId MISMATCH");
```
Convert the value to a Number for the compare function.